### PR TITLE
[proxmox_template] Fix error while uploading big ISO to Proxmox VE cluster

### LINKED ISSUE #5579: proxmox_template 'Connection aborted.'
### Summary

Proxmox cluster, all nodes running Proxmox 7.2-11, installed on Debian Bullseye

We have a playbook that uploads a custom ISO file using proxmox_template module and creates a new VM. We don't run it that often, yesterday we discovered it not working. The error from the task is:
```
fatal: [ht-dev-eh01.stg.isnic.is -> localhost]: FAILED! => changed=false 
  msg: 'uploading/downloading of template deploy-ht-dev-eh01.stg.isnic.is.iso failed with exception: (''Connection aborted.'', ConnectionResetError(54, ''Connection reset by peer''))'
```

This line appears in syslog on the Proxmox node:
```
pveproxy[1534092]: problem with client 2001:67c:6c:xxxx; No space left on device
```

There's plenty of space on all drives, both on the ansible controller and on the proxmox host. I've tracked the error to AnyEvent https://metacpan.org/release/MLEHMANN/AnyEvent-7.17/source/lib/AnyEvent/Handle.pm#L1360

It seems AnyEvent receives more than `rbuf_max` in it's read buffer, so it emits the confusing `Errno::ENOSPC` error.
Uploading ISO through Proxmox web UI works with the same user/pass as used in ansible playbook.

### Issue Type

Bug Report

### Component Name

proxmox_template

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.13.2]
  config file = /srv/devops/ansible/ansible.cfg
  configured module search path = ['/home/einar/.ansible/plugins/modules', '/usr/local/share/py39-ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.9/site-packages/ansible
  ansible collection location = /home/einar/.ansible/collections:/usr/local/share/py39-ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.9.15 (main, Nov 14 2022, 06:06:09) [Clang 13.0.0 (git@github.com:llvm/llvm-project.git llvmorg-13.0.0-0-gd7b669b3a
  jinja version = 3.1.2
  libyaml = True
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general
# /usr/local/lib/python3.9/site-packages/ansible_collections
Collection        Version
----------------- -------
community.general 5.3.0
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

Ansible controller is running FreeBSD 13.1-RELEASE p4
Remote host is running Proxmox 7.2-11 on Debian bullseye

### Steps to Reproduce

```yaml (paste below)
- name: "Upload generated deployment ISO to Proxmox (always returns 400 due to proxmox bug, failure ignored)"
  proxmox_template:
    api_host: "{{ proxmox_vm_api_name }}"
    api_user: "{{ proxmox_vm_api_user }}"
    api_password: "{{ proxmox_vm_api_pass }}"
    validate_certs: yes
    state: present
    node: "{{ machine_config['node'] }}"
    src: "{{ deployiso_dir }}/iso_generated/deploy-{{ inventory_hostname }}.iso"
    storage: "{{ proxmox_vm_storage_iso }}"
    content_type: "iso"
    force: yes # Overwrite old ISO
  when: proxmox_vm_secrets_available|default(false) and (proxmox_vm_iso_non_standard or config_result['changed'] or isoscript_result['changed'])
  retries: 3 # Proxmox sometimes fails with HTTP 400 error, workaround with retry. See is/nic/devops#13496.
  delay: 3
  ignore_errors: True # UGLY HACK: Proxmox returns 400 errors, but the ISO has been successfully uploaded. Just assume success for now. FIXME: Find out why this is.
  delegate_to: "localhost"
```

### Expected Results

I assumed the ISO would be uploaded to Proxmox template storage

### Actual Results

```console (paste below)
fatal: [ht-dev-eh01.stg.isnic.is -> localhost]: FAILED! => changed=false 
  msg: 'uploading/downloading of template deploy-ht-dev-eh01.stg.isnic.is.iso failed with exception: (''Connection aborted.'', ConnectionResetError(54, ''Connection reset by peer''))'
```

This line appears in syslog on the Proxmox node:
```
pveproxy[1534092]: problem with client 2001:67c:6c:xxxx; No space left on device
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by UnderGreen**
##### SUMMARY

Fixes #5579.
As per documentation from [proxmoxer](https://proxmoxer.github.io/docs/2.0/examples/files/) `requests_toolbelt` is required to upload large files.

Can't find the exact size of that **large file** definition, but from my personal testing on Darwin x86_64 it is around 260/261MB. Not sure where it is coming from, so can't explain it.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

proxmox_template

##### ADDITIONAL INFORMATION

Integration test output with updated module:

```
❯ ansible-test integration --allow-unsupported --local proxmox_template
Running proxmox_template integration test role

PLAY [testhost] ****************************************************************

TASK [Gathering Facts] *********************************************************
ok: [testhost]

TASK [proxmox_template : Create dummy ISO file] ********************************
changed: [testhost]

TASK [proxmox_template : Delete requests_toolbelt module if it is installed] ***
ok: [testhost]

TASK [proxmox_template : Install latest proxmoxer] *****************************
ok: [testhost]

TASK [proxmox_template : Upload ISO as template to Proxmox VE cluster should fail] ***
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: Failed to import the required Python library (requests_toolbelt) on santipov-ltm.internal.salesforce.com's Python /Users/s.antipov/src/ansible/collections/ansible_collections/community/general/.venv/bin/python3.11. Please read the module documentation and install it in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter
fatal: [testhost]: FAILED! => {"changed": false, "msg": "'requests_toolbelt' module is required to upload files larger than 256MB"}
...ignoring

TASK [proxmox_template : assert] ***********************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [proxmox_template : Install old (1.1.2) version of proxmoxer] *************
changed: [testhost]

TASK [proxmox_template : Upload ISO as template to Proxmox VE cluster should be successful] ***
changed: [testhost]

TASK [proxmox_template : assert] ***********************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [proxmox_template : Install latest proxmoxer] *****************************
changed: [testhost]

TASK [proxmox_template : Make smaller dummy file] ******************************
changed: [testhost]

TASK [proxmox_template : Upload ISO as template to Proxmox VE cluster should be successful] ***
changed: [testhost]

TASK [proxmox_template : assert] ***********************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [proxmox_template : Install requests_toolbelt] ****************************
changed: [testhost]

TASK [proxmox_template : Make big dummy file] **********************************
changed: [testhost]

TASK [proxmox_template : Upload ISO as template to Proxmox VE cluster should be successful] ***
changed: [testhost]

TASK [proxmox_template : assert] ***********************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [proxmox_template : Delete ISO file from host] ****************************
changed: [testhost]

PLAY RECAP *********************************************************************
testhost                   : ok=18   changed=10   unreachable=0    failed=0    skipped=0    rescued=0    ignored=1
```

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
One question: was requests_toolbelt already required in the past, i.e. did the module not work when it was around in all situations; or is this a new requirement? If it is a new requirement, this is a breaking change.

(If that's the case, it's probably better to warn about it missing when the size of the file is larger than a certain limit, instead of failing.)

Also I don't see how the toolbelt is being used. Is it used implicitly by proxmoxer or some other code in the module?

**Comment by UnderGreen** [comment]
@felixfontein looks like a problem with the `proxmoxer` module itself and I can reproduce it using plain proxmoxer. It is started with proxmoxer 1.2.0 and still failing in the master branch. Installing `requests_toolbelt` somehow resolves it(I am digging into it, but it looks like so far requesting another dependency is the way out).

**Comment by UnderGreen** [comment]
> Also I don't see how the toolbelt is being used. Is it used implicitly by proxmoxer or some other code in the module?

Yes, it is used in [proxmoxer](https://github.com/proxmoxer/proxmoxer/blob/develop/proxmoxer/backends/https.py#L215).

**Comment by UnderGreen** [comment]
@felixfontein I added notes about file size and usage of `requests_toolbelt`.

**Comment by felixfontein** [comment]
Hmm, I wasn't aware that the bot has write access to this repo. You could for now uncheck the checkmark that allows maintainers to push to your branch, and force push your version to get rid of the bot's commit (if you don't uncheck the checkmark, it will probably just push again).

**Comment by felixfontein** [comment]
I killed the config that the bot apparently uses in #6804, so it should no longer push unwanted commits.

Unfortunately it still makes CI fail by adding a failed check... I don't have sufficient privileges to get rid of the bot myself, but I've asked on IRC, hopefully it will be gone soon...

**Comment by felixfontein** [comment]
I've removed the commit; please now ignore the failing pre-commit.ci check :)

**Comment by webknjaz** [comment]
> do you know why, very spontaneously, I would say, the pre-commit.ci bot committed for my PR?

I requested the bot app install for the repos with the config having seen people running it in GHA in some subprojects which is inefficient, but didn't take into account that since it didn't run in some CIs the violations might've accumulated over time. I also forgot that by default it commits the changes to PRs. This one is solvable through a configuration change, but the bot would still report the check for as long as the config exists.

> Unfortunately it still makes CI fail by adding a failed check...

That shouldn't be happening. Sounds like a race condition, or maybe the PR wasn't rebased after that? Looks like it's the latter — the Checks widget says:
>It’s [1 commit](https://github.com/ansible-collections/community.general/compare/c4a2801f...main) behind (base commit: https://github.com/ansible-collections/community.general/commit/c4a2801f990f6b9486878bc6f789969107cd5fca)

**Comment by felixfontein** [comment]
> That shouldn't be happening. Sounds like a race condition, or maybe the PR wasn't rebased after that? Looks like it's the latter — the Checks widget says:

It also happened for a new PR (#6808) from `main` branch. I guess it somehow caches that a project should have a config.

File: tests/utils/constraints.txt [hunk 1]
```diff
@@ -57,3 +57,5 @@ redis ; python_version >= '3.6'
 pycdlib < 1.13.0 ; python_version < '3'  # 1.13.0 does not work with Python 2, while not declaring that
 python-daemon <= 2.3.0 ; python_version < '3'
 bcrypt < 4.0.0  # TEMP: restrict to < 4.0.0 since installing 4.0.0 fails on RHEL 8
+proxmoxer < 2.0.0 ; python_version <= '3.6'
+proxmoxer ; python_version > '3.6'
```
Comment (felixfontein) [inline tests/utils/constraints.txt:61]: That shouldn't be necessary.
```suggestion
```

**Comment by UnderGreen** [comment]
@felixfontein any insights on why it is still [failing](https://github.com/ansible-collections/community.general/actions/runs/5408402722/jobs/9827435288?pr=6757) for 2.6?
`importlib` is in requirements and see in logs it is downloaded(but not installed?).

**Comment by UnderGreen** [comment]
I skip 2.6 Python as it was never supported by `proxmoxer` library.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@UnderGreen thanks a lot for your contribution!

**Pull request merged**